src/VastManager.js

```diff
@@ -56,6 +61,102 @@ var vastManager = function () {
 		}
 	}
 
+	function showNextOverlay(show) {
+		var nextOverlays = document.getElementsByClassName('vjs-next-overlay');
+		if (nextOverlays && nextOverlays.length > 0) {
+			nextOverlays[0].style.display = show ? '' : 'none';
+		}
+	}
+
+	function needPlayAdForPlaylistItem(plIdx) {
+		if (_options.frequencyRules && _options.frequencyRules.playlistClips && _options.frequencyRules.playlistClips > 1) {
+			var mod = plIdx % _options.frequencyRules.playlistClips;
+			return mod === 0;
+		}
+		return true;
+	}
+
+	function nextListItemHandler() {
+		_nextPlaylistItemFired = true;
+		_savedMarkers = null;
+		showCover(true);
+		_playlistIdx++;
+		_contentDuration = 0;
+		_player.one('loadedmetadata', function() {
+			if (_markersHandler && _player.markers && _player.markers.destroy) {
+				_player.markers.destroy();
+			}
+			_contentDuration = parseInt(_player.duration()) - 0.5;
+			if (_playlistCreative) {
+				if (needPlayAdForPlaylistItem(_player.playlist.currentIndex())) {
+					if (_options.timeOffset != 'start') {
+						showCover(false);
+					}
+					play(_playlistCreative);
+					return;
+				}
+				showCover(false);
+				_logger.log(_prefix, 'Ad did not play due to frequency settings');
+				_player.playlist.autoadvance(0);
+			}
+			else {
+				if (!_prebidCommunicatorObj) {
+					_prebidCommunicatorObj = new _prebidCommunicator();
+				}
+				_prebidCommunicatorObj.doPrebid(_options, function(creative) {
+					_playlistCreative = creative;
+					if (creative) {
+						if (needPlayAdForPlaylistItem(_player.playlist.currentIndex())) {
+							if (_options.timeOffset != 'start') {
+								showCover(false);
+							}
+							play(_playlistCreative);
+						}
+						else {
+							showCover(false);
+							_logger.log(_prefix, 'Ad did not play due to frequency settings');
+							_player.playlist.autoadvance(0);
+						}
+					}
+					else {
+						showCover(false);
+						_player.playlist.autoadvance(0);
+					}
+				});
+				return;
+			}
+			if (_options.timeOffset != 'start') {
+				showCover(false);
+			}
+		});
+		setTimeout(function() {
+			if (_contentDuration === 0) {
+				showCover(false);
+			}
+		}, 1000);
+	}
+
+	function doPrebidForNextPlaylistItem() {
+		_playlist = (_player.playlist && typeof _player.playlist === 'function') ? _player.playlist() : [];
+		if (!_playlist) {
+			_playlist = [];
+		}
+		if (_playlist.length > 1 && _player.playlist.currentIndex() < _playlist.length - 1) {
+			if (!_playlistCreative) {
+				if (!_prebidCommunicatorObj) {
+					_prebidCommunicatorObj = new _prebidCommunicator();
+				}
+				_prebidCommunicatorObj.doPrebid(_options, function(creative) {
+					_playlistCreative = creative;
+					_player.playlist.autoadvance(!!_playlistCreative ? null : 0);
+				});
+			}
+		}
+		else {
+			_player.off('playlistitem', nextListItemHandler);
+		}
+	}
+
```

Comment:
Would be helpful to add comments for each of the new functions

Reply:
Added a few comments.